### PR TITLE
[GPU] UsmHostTensor implementation

### PR DESCRIPTION
**Pull request by sshlyapn**
### Details:
 - USMHostTensor implementation and some fixes

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/remote_tensor.cpp [hunk 1]
```diff
@@ -28,20 +40,24 @@ RemoteTensorImpl::RemoteTensorImpl(RemoteContextImpl::Ptr context,
...
+    : m_context(context)
+    , m_element_type(element_type)
+    , m_shape(shape)
+    , m_memory_object(memory)
+    , m_layout(make_layout(element_type, shape))
+    , m_mem_type(allocation_type_to_tensor_type(memory->get_allocation_type()))
+    , m_mem(nullptr)
+    , m_surf(0)
+    , m_plane(0) {
+    update_hash();
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/remote_tensor.cpp:60]: Aren't `update_stride()` and `update_properties()` needed here?

Comment (sshlyapn) [reply src/plugins/intel_gpu/src/plugin/remote_tensor.cpp:60]: Yeah, looks like they are needed. By the way, do you have any plans for this c-tr? Currently it's not used anywhere, maybe we can remove it at all?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/plugin/remote_tensor.cpp:60]: > By the way, do you have any plans for this c-tr?

I used it here https://github.com/vladimir-paramuzov/openvino/blob/dyn_memory_opt/src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp#L651 

Though you can remove it from this PR and we'll add it later if it's really needed.

Comment (sshlyapn) [reply src/plugins/intel_gpu/src/plugin/remote_tensor.cpp]: @vladimir-paramuzov , I see, thank you

I have removed it for now

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**